Mathlib/Algebra/Free.lean

```diff
@@ -479,7 +479,6 @@ variable {α : Type u}
 @[to_additive]
 instance : Semigroup (FreeSemigroup α) where
   mul L1 L2 := ⟨L1.1, L1.2 ++ L2.1 :: L2.2⟩
-  -- Porting note: replaced ext by FreeSemigroup.ext
   mul_assoc _L1 _L2 _L3 := FreeSemigroup.ext _ _ rfl <| List.append_assoc _ _ _
```

Comment:
This seems fine: `FreeSemigroup.ext` seems to be protected, and that seems as desired.

Reply:
Yes, I believe you're right. 

What do you think, @joelriou?

Reply:
I just double-checked this with mathlib3: it was indeed `free_semigroup.ext` that's used here in mathlib 3 so there is really nothing happening here.